### PR TITLE
Set Kernel config "Hardware Inited" to 1 (true)

### PR DESCRIPTION
**Pull request by mailwl**
pass through endless loop:
Batman Arkham Origins Blackgate - Won't boot #1610
Anpanman to Touch de Waku Waku Training

### CODE REVIEW

**Comment by KillzXGaming** [comment]
Some games i've tested with this with infinite loops now get fatal errors. 
Luigi's Mansion 2
[http://pastebin.com/aAaY72nT](LOG)

Mario Party Island Tour
[http://pastebin.com/RZUvVC1j](LOG)

**Comment by mailwl** [comment]
ic. well, i hope someone adds "dlp:SRVR" service to Citra

**Comment by KillzXGaming** [comment]
Wouldn't that be the download play client? Unsure why games would rely on this service.
https://www.3dbrew.org/wiki/DLP_Services

**Comment by mailwl** [comment]
yes, it is. but there is no sources in Citra for this service, so games, which try to access it, immediately breaks. but it's other issue, not related with this PR

**Comment by KillzXGaming** [comment]
Ok, feel free to delete those comments as they do not relate to the PR. I may add this as an issue.

**Comment by JohnGodgames** [comment]
This PR also helps with:

Bugs vs. Tanks!

**Comment by KillzXGaming** [comment]
This and
https://github.com/exhalatio/citra/tree/dlp_dummy
Fixed Luigi's Mansion 2
ediargz

**Comment by JohnGodgames** [comment]
@exhalatio can you make a PR?

I'm confirming https://github.com/exhalatio/citra/tree/dlp_dummy + this PR

marion island

mansion

**Comment by Leo626** [comment]
Rhythm Heaven The best+ now boots for me when additionally merged with exhalatio's [Dummy implementation dlp: SRVR Service. Only result success](https://github.com/exhalatio/citra/commit/8a689c9e730b19ed4808dd0028fe9d79813debc4) commit.

citra-qt_2016-03-29_20-43-47

**Comment by exhalatio** [comment]
@JohnGodgames
Hi, I make a PR it.

**Comment by MoochMcGee** [comment]
HACKS
A
C
K
S

**Comment by mailwl** [comment]
rename the variable to running_hw2

**Comment by JamePeng** [comment]
http://3dbrew.org/wiki/Configuration_Memory
I think it is totally wrong about the "running_hw2"

**Comment by mailwl** [comment]
hm... well, you right. it looks more like "Hardware Initialized"
```c++
int Anpanman_stuck()
{
  while ( !(v1FF81086 & 1) )
    sleep(1000000, 0);
  return v1FF81004;
}
```

**Comment by yuriks** [comment]
Where exactly did this "hardware inited" interpretation come from? There's absolutely nothing in the 3dbrew page that would suggest it, as far as I can tell.

**Comment by mailwl** [comment]
it's just my guess, from the code, where game Anpanman is stuck.
before get type of running hardware at 0x1FF81004, it waits until 0x1FF81086 become 1.

**Comment by bunnei** [comment]
Yeah, not a fan of the name "hardware_inited" if we don't actually know the purpose of this field... I'm concerned it could be taken as a 'known' and in fact be something totally different.

IMO, I'm okay with just calling this "unknown_foo" and adding a well-written comment in the code explaining what we know about it. I think this still falls in line with "stubbing" functionality rather than "hacking". Also might not hurt to open an issue about this (to do more testing and try to figure out what it is, or go back and update if/when 3dbrew documents it) and reference it in the comment.

**Comment by JayFoxRox** [comment]
There should be a hardware test which does the same code with a very short sleep (to force context switches if that is necessary) and logs the values (first to memory, later to file) of both registers.

This could indeed be a field which reports wether the OS already knows what kind of hardware it is running on (which is still not what `hardware_initialized` would suggest..). Maybe the value of v1FF81004 is bad while v1FF81086 is low. We should also try forcing these regs to different values to find out when the OS controls them (or static analysis - whatever floats your boat)

But I agree with bunnei - create a new issue for this and call the variable `unknown_...` until then.
Then get this merged asap unless it causes regressions.

**Comment by yuriks** [comment]
@bunnei Seems Good To Me

**Comment by bunnei** [comment]
@mailwl Please address the above comment and I think this can be merged!

**Comment by mailwl** [comment]
done

File: src/core/hle/shared_page.cpp [hunk 1]
```diff
@@ -16,6 +16,9 @@ void Init() {
     std::memset(&shared_page, 0, sizeof(shared_page));
 
     shared_page.running_hw = 0x1; // product
+
+    // Some games waits until this value become 0x1, before asking running_hw
```
Comment (JayFoxRox) [inline src/core/hle/shared_page.cpp]: (nit) ~ "// Some games wait until this value becomes 0x1, before reading 'running_hw'"

LGTM other than that

**Comment by mailwl** [comment]
done

**Comment by bunnei** [comment]
LGTM

**Pull request merged**